sys/parsoid.js

```diff
@@ -621,9 +621,7 @@ PSP.transformRevision = function(hyper, req, from, to) {
             original: original
         };
         if (from === 'sections') {
-            body2.html = {
-                body: replaceSections(original, parseSections(req))
-            };
+            body2.html = replaceSections(original, parseSections(req));
```

Comment:
Ah, you are right. `original` is framed as a pagebundle, but we historically included html as a non-pagebundle in the request. We need to change the section case to be consistent with the non-section case, and your change does just that. Evidently we don't have a test covering this. It would be really awesome if you added one as part of this PR. It should go into [transform.js](https://github.com/wikimedia/restbase/blob/master/test/features/parsoid/transform.js).

With the move towards separate data-mw and possibly other metadata in the future, I do wonder if the current ad-hoc format still makes sense, or whether we should convert the `modified` pagebundle to a proper pagebundle as well. This would make `original` and `modified` page structurally identical, and would afford both the same framing / header functionality. It would also be somewhat ironic if the pagebundle2pagebundle end point did not actually accept pagebundles.

In any case, we can address that question in a follow up. This change looks okay to me.